src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -178,100 +177,60 @@ private void checkExists(String path) {
-    }
+  private void connect() throws IOException {
+    org.apache.sshd.sftp.client.SftpClientFactory scf = org.apache.sshd.sftp.client.SftpClientFactory.instance();
+    sftp = scf.createSftpClient(session);
   }
 
-  private void configureSession(String user) throws JSchException {
-    Properties hash = new Properties();
-    configureHostChecking(hash);
-    setRandomPrng(hash);
-    if (!isEmpty(preferredAuthenticationMethods)) {
-      hash.put(PREFERRED_AUTHENTICATION_METHODS, preferredAuthenticationMethods);
-    }
+  private void configureSession(String user) throws IOException {
 
-    session = jsch.getSession(user, host);
-    session.setConfig(hash);
-    session.setPort(port);
-    session.setTimeout(Long.valueOf(connectionTimeoutMillis).intValue());
-    configureProxy(session);
-  }
+    session = client.connect(user, host, port)
+        .verify(connectionTimeoutMillis)
+        .getSession();
 
-  private void setRandomPrng(Properties hash) {
-    hash.put("random", prngAlgorithmClassImplementation);
-  }
+    session.setKeyIdentityProvider(KeyIdentityProvider.EMPTY_KEYS_PROVIDER);
 
-  private void configureHostChecking(Properties hash) throws JSchException {
-    if (knownHostsFile == null) {
-      hash.put(STRICT_HOST_KEY_CHECKING, "no");
-    } else {
-      checkExists(knownHostsFile);
-      hash.put(STRICT_HOST_KEY_CHECKING, "ask");
-      jsch.setKnownHosts(knownHostsFile);
-    }
+    configureProxy(session);
   }
 
-  private void configureProxy(Session session) {
+  private void configureProxy(ClientSession session) {
     if (proxyConfig != null) {
-
-      Proxy proxy = null;
+      InetSocketAddress proxyAddress = new InetSocketAddress(proxyConfig.getHost(), proxyConfig.getPort());
+      InetSocketAddress remoteAddress = new InetSocketAddress(this.host, this.port);
       switch (proxyConfig.getProtocol()) {
         case HTTP:
-          ProxyHTTP proxyHttp = new ProxyHTTP(proxyConfig.getHost(), proxyConfig.getPort());
-          if (proxyConfig.getUsername() != null && proxyConfig.getPassword() != null) {
-            proxyHttp.setUserPasswd(proxyConfig.getUsername(), proxyConfig.getPassword());
-          }
-          proxy = proxyHttp;
+          session.setClientProxyConnector(proxyConfig.getUsername() != null && proxyConfig.getPassword() != null
+              ? new HttpClientConnector(proxyAddress, remoteAddress, proxyConfig.getUsername(),
+                                        proxyConfig.getPassword().toCharArray())
+              : new HttpClientConnector(proxyAddress, remoteAddress));
           break;
-
-        case SOCKS4:
-          ProxySOCKS4 proxySocks4 = new ProxySOCKS4(proxyConfig.getHost(), proxyConfig.getPort());
-          if (proxyConfig.getUsername() != null && proxyConfig.getPassword() != null) {
-            proxySocks4.setUserPasswd(proxyConfig.getUsername(), proxyConfig.getPassword());
-          }
-          proxy = proxySocks4;
-          break;
-
         case SOCKS5:
```

Comment:
@tomastoloza I appeal to your memory, why don't we have SOCKS4 in the case? At the very least, shouldn't we declare it to be resolved with the same code as SOCKS5?